### DISCUSSION TITLE
Proper Print Styling for Listing and Detail Pages

### DISCUSSION THREAD

**Discussion by Fusseldieb** [directus/directus#2972] [Feature Requests]
Hey!

One thing that I miss in Directus is the ability to print a document (item in a collection) that's currently open.
If I try to `Ctrl+P`, it just prints out the mobile view of the page, which (in my case with +- 30 fields) spans over 5-6 pages.

Currently Directus has 2 `.vue` files for each interface - An `Edit` and a `Detail` component. Currently it just takes a "screenshot" of the page and obviously uses the `Edit` component, which is considerably big in physical size, however, the `Detail` component saves the most space in my opinion, so it should also be OK for printing. If not, maybe adding a third `Print` component could be useful. 
Another approach would be to load a customizable template defined by the user like:
```
Size:
{{size}}

Height:
{{height}}

...
```
To make it customizable there are great libraries such as Editorjs - or even simple Markdown - or maybe even the WYSIWYG interface already present in Directus (As it seems it already generates HTML).

I'm just throwing some ideas out there...

Also, since I have moderate-to-good experience with Vue, I could help develop that feature and would love to contribute and/or see it being integrated into Directus.

**Comment by benhaynes** [comment]
I love this idea. It will require some serious fine-tuning, but it shouldn't be too difficult. We'll need to keep the print _relatively_ true to the page design though... since we can't predict what/how a custom interface or display might want to render data (and raw values might be useless). That said, I think we can optimize what is displayed, and give a very clean print option for listing and detail pages.

**Reply by Fusseldieb** [reply]
Would be absolutely awesome! If I can help in anything, let me know! I'm currently searching some free time to help to make some PRs for this - that is - if I master the inner workings of Directus. I have never contributed to a Vue project this size.

**Reply by benhaynes** [reply]
That sounds great! Take the codebase for a spin, and see if it all makes sense. I can mockup a design for the listing/detail pages and then you can give a PR a try if you have time/interest. 👍

**Reply by Fusseldieb** [reply]
Cool! I will definitively do that! I'll let you know as soon as I'm more confortable with the codebase. Meanwhile, I'll play with it and try to do the basics of this implementation.

**Reply by gearoidc** [reply]
Any updates on this feature? Would be great to have.

**Reply by Fusseldieb** [reply]
I'm still playing around with Directus @gearoidc , from **me** there is (unfortunately) no update. Maybe someone else did something interesting? 😄

**Reply by rijkvanzanten** [reply]
No updates here! I'm wondering if we should handle this on the CSS in the app, or if we should focus on having an "export to PDF" (https://github.com/directus/directus/discussions/5642) instead 🤔

**Reply by benhaynes** [reply]
I think it would be nice to have a `@media print` built-in that could be used to clean up the App for printing (both to paper, and to PDF). I would love to draft a design for this.

**Reply by alexsonphoenix** [reply]
Any new updates on this? This is such a great feature to have in  Directus

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
